pkg/runner/job_runner.go

```diff
@@ -256,10 +271,54 @@ func (jr *JobRunner) Run(j *job.Job) ([][]*job.Report, []*job.Report, error) {
 	return allRunsReports, finalReports, nil
 }
 
+// GetCurrentRun returns the run which is currently being executed
+func (jr *JobRunner) GetCurrentRun(jobID types.JobID) (*types.RunID, error) {
+	runEvents, err := jr.eventManager.Fetch(
+		[]frameworkevent.QueryField{
+			frameworkevent.QueryJobID(jobID),
+			frameworkevent.QueryEventName(EventRunStarted),
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch last run id for job %d: %v", jobID, err)
+	}
+
+	lastEvent := runEvents[len(runEvents)-1]
+	payload := runStartedPayload{}
+	if err := json.Unmarshal([]byte(*lastEvent.Payload), &payload); err != nil {
+		return nil, fmt.Errorf("could not fetch last run id for job %d: %v", jobID, err)
+	}
+	return &payload.RunID, nil
+
+}
+
+func (jr *JobRunner) emitEvent(jobID types.JobID, eventName event.Name, payload interface{}) error {
+	payloadJSON, err := json.Marshal(payload)
+	if err != nil {
+		jobLog.Warningf("could not encode payload for event %s: %v", eventName, err)
+		return err
+	}
+
+	rawPayload := json.RawMessage(payloadJSON)
+	ev := frameworkevent.Event{JobID: jobID, EventName: eventName, Payload: &rawPayload, EmitTime: time.Now()}
```

Comment:
Just a note:

May be we should consider to use `time.Now().UTC()` instead of `time.Now()` through the whole ConTest. It will prevent possible problems with timezones.

Reply:
The object returned by `Time.Now()` has timezone information embedded (and then we store in UTC)? And as far as I know there is no concept of naive time object in Go (contrary to Python), so time comparison should always be done between objects with timezone information.

Reply:
the only risk could be related to storing/retrieving the time string to/from database, but we store the non-naive timestamp so this should never become an issue. I agree that UTC is better though, it could be done in the future if we identify more important gaps related to this

Reply:
> And as far as I know there is no concept of naive time object in Go (contrary to Python), so time comparison should always be done between objects with timezone information.

"Possible problems with timezones" that I referred to -- are problems while converting time to different data-formats (go:Time -> JSON -> DB -> go:Time and so on).

Never mind. It was just a (may be stupid) thought. I just used to think that timezones is a sources of vague bugs.

Reply:
We store in `TIMESTAMP` MySQL format which interprets incoming time object as belonging to  the timezone which is set for the current connection (which is the client timezone). It's then persisted as UTC and returned again in the timezone set within the connection for the query. I'd need to look into how the connector works if the incoming time object is encoded in a timezone which does not match the connection set in the timezone (I assume it's handled transparently, but I'd need to confirm).

As for encoding/decoding, we would still serialize with timezone information (e.g. RFC3339) and de-serialize accordingly. I won't address this now, but let's revisit if we really think this approach is fragile.

Reply:
Time zones are a source of HUGE problems, not just vague, and you're right to worry about them :) What we mean here is that the timestamps are already stored in a non-naive way, which eliminates (minimizes?) the risk of timezone-related problems. Additionally, Go makes it really hard to work with naive timestamps, which helps keep our sanity